src/test/java/com/amihaiemil/camel/NoCommentsYamlLineTest.java

```diff
@@ -99,18 +99,18 @@ public void trimsCommentsLine() {
     @Test
     public void trimsCommentsAtEndOfLine() {
         YamlLine noComments = new NoCommentsYamlLine(
-            new RtYamlLine("  this isn't comment   #here is the comment", 1)
+            new RtYamlLine("  \"this isn't comment\"   #here is the comment", 1)
```

Comment:
@SherifWaly make this test as it was :D 

This test failed with the ``else`` change, because it actually found a bug.
You changed the test to pass, instead of fixing the bug -- it's a common mistake, I also used to do it, not realizing that the code should be changed to pass the test, not the other way around (of course, assuming the test is well written, which this test was)